core/auth.py

```
@@ -0,0 +1,697 @@
+        keys_data = {kid: key.to_dict() for kid, key in self._api_keys.items()}
+        json_data = json.dumps(keys_data).encode()
+        encrypted_data = self._fernet.encrypt(json_data)
+
+        with open(API_KEYS_FILE, 'wb') as f:
+            f.write(encrypted_data)
+
+    def _get_jwt_secret(self) -> str:
+        """Get JWT secret key from secure secrets storage."""
+        try:
+            # Try to get from secrets manager first
+            secret = get_secret("jwt_secret_key")
```

Ruff (0.14.10)

Undefined name `get_secret`

(F821)